main.cpp

```diff
@@ -49,77 +49,105 @@ Create a branch named Part1
  
  Send me the the link to your repl.it in a DM on Slack
 
- Wait for my code review.
+ Wait for my code review...
  */
 
 #include <iostream>
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    float value;
+    const std::string name; 
+    
+    T (float v, const char* aCharacter) :
+        value (v),
+        name (aCharacter)
+    { }
 };
 
-struct <#structName1#>                                //4
+struct Operation 
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
-    {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
-        return nullptr;
-    }
+    T* compare(T* a, T* b)
+    {   
+        if (a != nullptr && b != nullptr) 
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
+        return nullptr; 
+    } 
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float var1 { 0 }, var2 { 0 };
+    
+    U(float a) : var1 (a) { }
+    
+    U();
+    
+    float product( float* inputValue ) 
     {
-        
+        std::cout << "\n(non-static) U's var1 value: " << this->var1 << std::endl;
+        if (inputValue != nullptr) { this->var1 = *inputValue;}
+        std::cout << "\n(non-static) U's var1 updated value: " << this->var1 << std::endl;
+        std::cout << "Reducing magnitude.. \n";
+        while( std::abs (static_cast <long> (this->var2 - this->var1)) > 0.001f )
+        {
+            /*
+             write something that makes the distance between that->name2 and that->name1 get smaller
+             */
+            this->var2 += 0.5f;
+            std::cout << this->var2 << ", ";
+
+        }
+        std::cout << "\n(non-static) U's var2 updated value: " << this->var2 << std::endl;
+        return this->var2 * this->var1;
```

Comment:
I'm more looking for you to only do the work with this if `inputValue` is not `nullptr`.   You're still doing the work regardless.  Can you adjust?

Reply:
Yes, I see... fixed.